R/augsynth_class.R

```diff
@@ -0,0 +1,293 @@
+                            errors_se = augsynth$lambda_errors_se)
+        p <- ggplot2::ggplot(errors, ggplot2::aes(x = lambdas, y = errors)) +
+            ggplot2::geom_point(size = 2) +
+            ggplot2::geom_errorbar(
+                ggplot2::aes(ymin = errors,
+                             ymax = errors + errors_se),
+                width=0.2, size = 0.5)
+        p <- p + ggplot2::labs(title = bquote("Cross Validation MSE over " ~ lambda),
+                               x = expression(lambda), y = "Cross Validation MSE",
+                               parse = TRUE)
+        p <- p + ggplot2::scale_x_log10()
+
+        # find minimum and min + 1se lambda to plot
+        min_lambda <- choose_lambda(augsynth$lambdas,
+                                    augsynth$lambda_errors,
+                                    augsynth$lambda_errors_se,
+                                    F)
+        min_1se_lambda <- choose_lambda(augsynth$lambdas,
+                                        augsynth$lambda_errors,
+                                        augsynth$lambda_errors_se,
+                                        T)
+        min_lambda_index <- which(augsynth$lambdas == min_lambda)
+        min_1se_lambda_index <- which(augsynth$lambdas == min_1se_lambda)
+
+        p <- p + ggplot2::geom_point(
+            ggplot2::aes(x = min_lambda,
+                         y = augsynth$lambda_errors[min_lambda_index]),
+            color = "gold")
+        p + ggplot2::geom_point(
+            ggplot2::aes(x = min_1se_lambda,
+                         y = augsynth$lambda_errors[min_1se_lambda_index]),
+            color = "gold") +
+            ggplot2::theme_bw()
+        return(p)
+    } else if (plot_type == 'estimate only') {
+        p <- augsynth_plot_from_results(augsynth, inf_type = 'none')
+    } else if (plot_type == 'estimate') {
+        p <- augsynth_plot_from_results(augsynth, inf_type = inf_type)
+    } else if (grepl('placebo', plot_type)) {
+        p <- permutation_plot(augsynth, inf_type = inf_type)
+    } else if (plot_type == 'outcomes') {
+        p <- augsynth_outcomes_plot(augsynth, measure = 'synth')
+    } else if (plot_type == 'outcomes raw average') {
+        p <- augsynth_outcomes_plot(augsynth, measure = c('synth', 'average'))
+    }
+    return(p)
+}
+
+
+
+
+#' Methods for accessing details of augsynth result object (of class augsynth)
+#'
+#'
+#' @param x augsynth result object
+#'
+#' @rdname augsynth_class
+#'
+#' @return is.augsynth: TRUE if object is a augsynth object.
+#'
+#' @export
+is.augsynth <- function(x) {
+    inherits(x, "augsynth")
+}
+
+
+
+#'
+#' @return dim: Dimension of data as pair of (# units, # time points).
+#'
+#' @rdname augsynth_class
+#' @export
+#'
+dim.augsynth <- function(x, ... ) {
+    n_unit = length( unique( x$raw_data[[ x$unit_var ]] ) )
+    n_time = length( unique( x$raw_data[[ x$time_var ]] ) )
+    return( c( n_unit, n_time ) )
+}
+
```

Comment:
I think the number of X units should be the same using `x$raw_data` and `nrow(augsynth$data$X)`. But does `augsynth$data$y` only keep post-treatment time periods? 

Another reason to hold onto `augsynth$raw_data` is so that we can run `update_augsynth()` without needing to pass in the original unprocessed data again.